platform/darwin/src/MGLOfflineStorage.h

```diff
@@ -174,6 +184,21 @@ MGL_EXPORT
  */
 @property (class, nonatomic, readonly) MGLOfflineStorage *sharedOfflineStorage;
 
+#pragma mark - Adding Contents of File
+
+/**
+ Adds the contents of the file path.
+ 
+ Once the content is added, then the `completion` block is executed with the
+ packs passed in.
+ 
+ @param filePath A string representation of the file path. The file path must be
+ writable as schema updates may be perfomed.
+ @param completion The completion handler to call once the file has been added.
+ This handler is executed asynchronously on the main queue.
+ */
+- (void)addContentesOfFile:(NSString *)filePath withCompletionHandler:(nullable MGLOfflinePacksAdditionCompletionHandler)completion;
```

Comment:
For Swift, we should also add `-addContentsOfURL:withCompletionHandler:` alongside this method. Swift strongly prefers using URLs for file paths.